std/range/primitives.d

```diff
@@ -2213,7 +2213,7 @@ equivalent to `popFront(array)`. For $(GLOSSARY narrow strings),
 `popFront` automatically advances to the next $(GLOSSARY code
 point).
 */
-void popFront(T)(ref T[] a) @safe pure nothrow @nogc
+void popFront(T)(scope ref inout(T)[] a) @safe pure nothrow @nogc
```

Comment:
`inout` makes no sense if the function isn't returning anything. I'm honestly surprised that it even compiles. Why add it? The same goes for the various other pop functions below.

Reply:
It means "you can use the exact same compiled function regardless of the storage class of `a`'s elements."  It results in fewer distinct functions in the final executable.  See https://github.com/dlang/phobos/pull/6743#issuecomment-435770952.

Reply:
Why not just `const(T)[]` then?  `inout` is basically const inside the function body, the only difference is that the return value may be implicitly converted to either immutable or mutable since the function cannot have modified it.  Since nothing is being returned here, `inout` is misleading. Just use `const` instead.

Reply:
Argh... I see why now.  The compiler sees `const(T)` and tries to match *that* to the parameter type, rather than matching `T` to the parameter type first then implicitly converting to `const(T)`.  I'm tempted to call this a compiler bug / language defect, but I'm not sure I want to open the can of worms that is how IFTI performs its black magic.

As for why `inout` works at all, I'm not sure I want to know the real reason. But I guess it will have to suffice for now, so that this PR doesn't devolve into a tarpit of how/whether/when the compiler could be fixed to allow passing `T` to `const(T)` in a template parameter.